app/services/sendgrid_api.rb

```diff
@@ -0,0 +1,56 @@
+class SendgridApi
+  extend SingleForwardable
+
+  def_single_delegators :new, :spam_reported?, :bounced?, :smtp_alive?
+
+  def spam_reported?(email)
```

Comment:
There definitely *is* duplication. I didn't think it was worth trying to extract any commonality right now, because I'm expecting to add a couple more functions (to cancel bounces and spam reports) that will have slightly different behaviour.